run-tox-env.yml

```diff
@@ -87,6 +87,14 @@ jobs:
           - bash: docker run -v $PWD:/project -i --name ${{ variables.docker_name }} -d ${{ variables.docker_image }} /bin/bash
             displayName: Start up Docker container
 
+        - ${{ if parameters.cache_dir }}:
+          - bash: mkdir ${{ parameters.cache_dir }}
+            displayName: Create cache directory
+          - task: Cache@2
+            inputs:
+              key: '${{ parameters.cache_dir }}/*'
```

Comment:
Maybe making the key not a file path will allow it to restore without the files existing. Perhaps we should set this to the job display name.

Reply:
The key needs to be an identifier that changes when the cache needs to be updated, hence why it is currently a glob of all the files under the cache directory.